File: backend/restaurant-system/src/main/java/com/restaurant/restaurantsystem/entity/Order.java
package com.restaurant.restaurantsystem.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "orders")
public class Order extends CommonData{
    @Column(name = "image")
    private String img;
    @Column(name = "price")
    private double price;
    @Column(name = "description")
    @Lob
    private String description;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "category_id")
    @JsonIgnore
    private Category category;

}

File: frontend/src/app/model/order.ts
export class Order {
    id!: number;
    name!: string;
    createdDate!: Date;
    updatedDate!: Date;
    img!:string;
    price!: number;
    description!: string;

    constructor(id:number,name:string,
        createdDate:Date,updatedDate:Date,
        img:string,price:number,
        description:string){
            this.id=id;
            this.name=name;
            this.createdDate=createdDate;
            this.updatedDate=updatedDate;
            this.img=img;
            this.price=price;
            this.description=description;
        }
}
